modules/ding_frontend/ding_frontend.admin.inc

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * @file
+ * Module's admin features.
+ */
+
+/**
+ * Administration form.
+ */
+function ding_frontend_settings($form, &$form_state) {
+  $types = array(
+    'ding_news' => t('News'),
+    'ding_event' => t('Event'),
+  );
+  $form['ding_frontend_fe_promote_types'] = array(
+    '#title' => t('Promote types to Fronpage'),
+    '#type' => 'checkboxes',
+    '#options' => $types,
+    '#default_value' => variable_get('ding_frontend_fe_promote_types', array('ding_event' => 'ding_event')),
+  );
+
+  $form = system_settings_form($form);
+  $form['#submit'][] = 'ding_frontend_settings_submit';
+  return $form;
+}
+
+/**
+ * Custom submit handler for settings form.
+ */
+function ding_frontend_settings_submit($form, &$form_state) {
+  $types = array(
+    'ding_news',
+    'ding_event',
+  );
+  foreach ($types as $type) {
+    $conf = variable_get('node_options_' . $type, array('status', 'promote'));
+    if (in_array('promote', $conf)) {
+      unset($conf[array_search('promote', $conf)]);
+    }
+    if ($form_state['values']['ding_frontend_fe_promote_types'][$type]) {
+      $conf[] = 'promote';
+    }
+    variable_set('node_options_' . $type, $conf);
```

Comment:
I think we misunderstand each other.

As I understand this change results in the following use:

1. By default events are promoted to frontpage. News are not. [This](https://github.com/ding2/ding2/blob/master/modules/ding_event/ding_event.strongarm.inc#L139-L147) [is](https://github.com/ding2/ding2/blob/master/modules/ding_news/ding_news.strongarm.inc) reflected in Features.
2. A local administrator wants to make news promoted to frontpage by default as well.
3. She uses the settings form provided here. This default values in the settings form is that *Events* are ticket off. This is derived from the `ding_frontend_fe_promote_types` variable. The exported value of this allows frontpage promotion of events - not news.
4. She submits the form with both *Events* and *News* ticked off.
5. Drupal adds the `promote` option to the `node_options_ding_news` variable. `node_options_ding_event` already has it. The value of the `ding_frontend_fe_promote_types` variable is not set.
6. When she revisits the form only *Events* will be ticked off. **The default value is derived from the variable `ding_frontend_fe_promote_types` which has not been updated in step 5. She expected both *Events* and *News* to be ticked off**.
7. When a new version of DDB CMS is deployed system administrators will revert all features. **This includes `ding_frontend_fe_promote_types` (which is not changed), `node_options_ding_news` and `node_options_ding_events`. This means that news will no longer be promoted to frontpage by default**
8. The local administrator can go back to step 1.

My suggestion:

1. By default events are promoted to frontpage. News are not. This is reflected by setting variables `node_options_ding_event` and `node_options_ding_news` accordingly in `ding_event_install()` and `ding_news_install()`.
2. A local administrator wants to make news promoted to frontpage by default as well.
3. She uses the settings form provided here. This default values in the settings form is that *Events* are ticket off. This is derived by inspecting the value of the `node_options_ding_event` and `node_options_ding_news` variables.
4. She submits the form with both *Events* and *News* ticked off.
5. Drupal adds the `promote` option to the `node_options_ding_news` variable. `node_options_ding_event` already has it.
6. When she revisits the form both *News* and *Events* will be ticked off as **the default value is derived by inspecting the value of the `node_options_ding_event` and `node_options_ding_news` variables**.
7. When a new version of DDB CMS is deployed system administrators will revert all features. **This does not include `node_options_ding_news` and `node_options_ding_events` as they are not included in Features**.
8. The local administrator does not have to do anything in this regard.